homeassistant/components/camera/foscam.py

```diff
@@ -78,3 +82,63 @@ def camera_image(self):
     def name(self):
         """Return the name of this camera."""
         return self._name
+
+    def set_motion_status(self, mode):
+        """Post to the camera and enable motion detection."""
+        if mode == CAMERA_ARMED:
+            enabled = '1'
+        else:
+            enabled = '0'
+
+        # Fill the URI with the command to enable motion detection
+        # Along with that as per foscam spec we have to set
+        # sensitivity: how much sensitivity camera should detect
+        # trigger interval: interval between each motion triggers
+        # schedule: set schedule for days. default is 24x7
+        # area: sort of like zones. default is all areas visible to cam
+        uri_template = self._base_url \
```

Comment:
This is too much protocol specific information inside Home Assistant. Please use an existing 3rd party lib to control Foscam or create a new one.